backup/wrappers.go

```diff
@@ -772,11 +772,22 @@ func backupExtendedStatistic(metadataFile *utils.FileWithByteCount) {
  */
 
 func backupTableStatistics(statisticsFile *utils.FileWithByteCount, tables []Table) {
-	attStats := GetAttributeStatistics(connectionPool, tables)
-	tupleStats := GetTupleStatistics(connectionPool, tables)
-
 	backupSessionGUC(statisticsFile)
-	PrintStatisticsStatements(statisticsFile, globalTOC, tables, attStats, tupleStats)
+
+	length := len(tables)
+	slice := length / MustGetFlagInt(options.WITH_STATS)
+	if slice < 1 {
+		slice = 1
+	}
+	for start := 0; start < length; start += slice {
+		end := start + slice
+		if end > length {
+			end = length
+		}
+		attStats := GetAttributeStatistics(connectionPool, tables[start:end])
+		tupleStats := GetTupleStatistics(connectionPool, tables[start:end])
+		PrintStatisticsStatements(statisticsFile, globalTOC, tables[start:end], attStats, tupleStats)
+	}
```

Comment:
Here some thoughts about this solution.

- This appproach leads to sending numerous separate queries to the connection.
- If you start to modify gpbackup options logic, you should document the concept of batch, and provide guide for optimal batch size calculation.

- Moreover, we could propbably aviod filling  `map[uint32][]AttributeStatistic`  `map[uint32]TupleStatistic` during statistics backup, and extract statistics from database  **cursor** via `connectionPool.Query` or `connectionPool.Queryx`. And make straight backup without intermediate maps

Reply:
> * This approach leads to sending numerous separate queries to the connection.

This is the main goal of this solution to reduce memory consumption.

Reply:
> * we could probably aviod filling  `map[uint32][]AttributeStatistic`  `map[uint32]TupleStatistic` during statistics backup, and extract statistics from database  **cursor** via `connectionPool.Query` or `connectionPool.Queryx`. And make straight backup without intermediate maps

Yes, but it will be a more global refactoring

Reply:
> * you should document the concept of batch

There is no specific documentation in the gpbackup repository

Reply:
> This is the main goal of this solution to reduce memory consumption.

That's approach increases maintenance costs, at least in terms of planning.

>Yes, but it will be a more global refactoring

Not really, there is not much code in `backupTableStatistics` logic.  We can still use `AttributeStatistic` and `TupleStatistic` structs when working with cursors, but use them directly, for example in new alternative `PrintStatisticsStatements` function

Reply:
> There is no specific documentation in the gpbackup repository

Anyway it should be described somewhere,  customers should be informed somehow

Reply:
rewrited to backup statistics in batches by 10000 tables at once